pkg/kubelet/dockershim/docker_service.go

```diff
@@ -524,6 +526,42 @@ func (d *dockerLegacyService) GetContainerLogs(pod *v1.Pod, containerID kubecont
 	return d.client.Logs(containerID.ID, opts, sopts)
 }
 
+// LegacyLogProvider gives the ability to use unsupported docker log drivers (e.g. journald)
+type LegacyLogProvider interface {
+	// Get the last few lines of the logs for a specific container.
+	GetContainerLogTail(uid kubetypes.UID, name, namespace string, containerID kubecontainer.ContainerID) (string, error)
+}
+
+// legacyLogProvider implements the LegacyLogProvider interface
+type legacyLogProvider struct {
+	dls DockerLegacyService
+}
+
+func NewLegacyLogProvider(dls DockerLegacyService) LegacyLogProvider {
```

Comment:
you can keep the implementation here.  if it causes a cycle, i could see an argument to move it in pkg/kubelet, but hopefully that is not the case.